diff --git a/include/configs/egf_3sm2008.h b/include/configs/egf_3sm2008.h
index e92099139d..6d35a7884b 100644
--- a/include/configs/egf_3sm2008.h
+++ b/include/configs/egf_3sm2008.h
@@ -119,7 +119,7 @@
 	"mmcpart=1\0" \
 	"mmcroot=/dev/mmcblk2p2 rootwait rw quiet \0" \
 	"mmcautodetect=yes\0" \
-	"mmcargs=setenv bootargs ${jh_clk} ${mcore_clk} console=${console} root=${mmcroot}\0 " \
+	"mmcargs=setenv bootargs ${jh_clk} ${mcore_clk} console=${console} root=${mmcroot} vt.global_cursor_default=0\0 " \
 	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bsp_script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
